wallet/src/Pos/Wallet/Web/Tracking/Sync.hs

```diff
@@ -478,13 +471,14 @@ rollbackModifierFromWallet wid newTip CAccModifier{..} = do
     WS.setWalletSyncTip wid newTip
 
 evalChange
-    :: [CWAddressMeta] -- ^ All adresses
-    -> [CId Addr]      -- ^ Own input addresses of tx
-    -> [CId Addr]      -- ^ Own outputs addresses of tx
+    :: [CWAddressMeta]  -- ^ Own input addresses of tx
+    -> [CWAddressMeta]  -- ^ Own outputs addresses of tx
     -> [CId Addr]
-evalChange allAddresses inputs outputs
-    | null inputs || null outputs = []
-    | otherwise = filter (isTxLocalAddress allAddresses (NE.fromList inputs)) outputs
+evalChange inputs outputs
+    | [] <- inputs = []
+    | inp : _ <- inputs =
+        let srcAccount = addrMetaToAccount inp
```

Comment:
oh, I realized that plan was great but didn't consider transactions which we send from account to the same account. now we can't differ change address from destination address.

on the other hand, AFAIU we can get all addresses of account and check each of output addresses. what do you think?

Reply:
Actually this problem exists for a long time, and we didn't care much. For `A->A` transactions we suggested that all its outputs are changes. And now processing is the same.
(Btw now I wonder, with such logic, don't we remove extra change addresses on rollback?)

Reply:
>Actually this problem exists for a long time

why? we took our addressed and determined change address as "not presented in our addresses".
and I suggest doing a similar thing here.

yes, there will still cases when we are wrong, but it will be ok for sending from one account to the same account. not it's broken.

>Btw now I wonder, with such logic, don't we remove extra change addresses on rollback?

hm, IIRC we were discussing it and decided to do 
`type CustomAddresses = HashMap (CId Addr) HeaderHash`
so we will remove it only when block which added this address will be rolled back.